src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/AndInstruction.cs

```diff
@@ -159,13 +159,13 @@ public override int Run(InterpretedFrame frame)
                     }
                     else
                     {
-                        frame.Push((bool)right ? null : ScriptingRuntimeHelpers.Boolean_False);
+                        frame.Push((bool)right ? null : Utils.BoxedFalse);
```

Comment:
There are certainly times when `frame.Push()` gets called with other boxed bools, such as if a method-call returned one, and checking for them at runtime is almost certainly more expensive than it saves.